### ISSUE TITLE
":exact" is not supported in Search

### ISSUE THREAD

**Issue by Efreeto** [Microsoft/fhir-server#305]
**Describe the bug**
":exact" is not supported in Search

**To Reproduce**
1. Create and run a working query with strings such as "http://xyz.com/Patient?address-city=Seattle&family=Smith"
2. Run a query with ":exact" such as "http://xyz.com/Patient?address-city:exact=Seattle&family=Smith"

**Expected behavior**
Search is performed regularly as in "https://r3.smarthealthit.org/Patient?address-city:exact=Seattle" and displays an empty result if there is no match

**Actual behavior**
An error is shown "Modifier 'Exact' is not supported for search parameter '_tag'." regardless of the search.

**Comment by hansenms** [comment]
Investigating. @jackliums

**Comment by jackliums** [comment]
:exact should be supported on string search parameters.

Based on the error message, it seems like you are trying to supply :exact modifier on _tag search parameter which is a token search parameter not string search parameter.

Can you check to make sure the modifier is actually being supplied on the string search parameter?

**Comment by Efreeto** [comment]
@jackliums Yes you are right. `exact` only does not work with _tag, but other string searches seem to work fine. I had thought those were the same. I will edit the original comment.

**Comment by jackliums** [comment]
@Efreeto Based on the spec (http://hl7.org/fhir/search.html#modifiers), I don't believe :exact is supported for token search parameter. I did a quick search but couldn't find anything indicating that should be the case?

**Comment by Efreeto** [comment]
@jackliums I think you are absolutely right. Nowhere in the document says it should be supported. I did a few tests on other servers, and it looks `:exact` is not supported because it is assumed for token searches anyway. 

However the other fhir servers I tried `:exact` handled it by quietly ignoring the parameter, even for incorrect parameters like `:xact'. They did not show me errors like azure fhir server did, and this made us think that this was a bug on azure fhir server. I apologize for this mistake, and please feel free to close this.

And let me know if you want me to file a bug to handle the error like other servers do (i.e. ignore wrong parameters), but I don't know if you even need this.
